CMakeLists.txt

```diff
@@ -140,3 +141,9 @@ endif()
 
 # Set up the packaging of the project.
 include( algebra-plugins-packaging )
+
+# Set GPU-related options
+if(ALGEBRA_PLUGIN_BUILD_CUDA)
+  set( CMAKE_CUDA_STANDARD 17 CACHE STRING "The (CUDA) C++ standard to use" )
+  enable_language(CUDA)
+endif()
```

Comment:
The `ALGEBRA_PLUGIN_BUILD_CUDA` variable is not used for anything at the moment... You didn't add any CUDA source files to the project. Making use of the modified headers "only" requires the clients of the `INTERFACE` libraries to set up their own builds with CUDA correctly.

I would not make any changes to this file...